File: app/src/main/java/org/qp/android/questopiabundle/lib/LibIProxy.java
package org.qp.android.questopiabundle.lib;

import android.net.Uri;

import androidx.documentfile.provider.DocumentFile;

import org.qp.android.questopiabundle.GameInterface;

public interface LibIProxy {
    /**
     * Starts the library thread.
     */
    void startLibThread();

    /**
     * Stops the library thread.
     */
    void stopLibThread();

    void enableDebugMode(boolean isDebug);

    void runGame(long gameId, String gameTitle, Uri gameDirUri, Uri gameFileUri);

    void restartGame();

    void loadGameState(Uri uri);

    void saveGameState(Uri uri);

    void onActionClicked(int index);

    void onObjectSelected(int index);

    void onInputAreaClicked();

    void onUseExecutorString();

    /**
     * Starts execution of the specified line of code in the library.
     */
    void execute(String code);

    /**
     * Starts processing the location counter in the library.
     */
    void executeCounter();

    LibGameState getGameState();

    void setGameInterface(GameInterface view);

}

File: app/src/main/java/org/qp/android/questopiabundle/lib/LibIProxy.kt
package org.qp.android.questopiabundle.lib

import android.net.Uri
import kotlinx.coroutines.flow.MutableSharedFlow
import org.qp.android.questopiabundle.GameInterface
import org.qp.android.questopiabundle.LibReturnValue
import org.qp.android.questopiabundle.dto.LibGameState

interface LibIProxy {
    /**
     * Starts the library thread.
     */
    fun startLibThread()
    /**
     * Stops the library thread.
     */
    fun stopLibThread()
    fun enableDebugMode(isDebug: Boolean)
    fun runGame(gameId: Long, gameTitle: String, gameDirUri: Uri, gameFileUri: Uri)
    fun restartGame()
    fun loadGameState(uri: Uri)
    fun saveGameState(uri: Uri)
    fun onActionClicked(index: Int)
    fun onObjectSelected(index: Int)
    fun onInputAreaClicked()
    fun onUseExecutorString()
    /**
     * Starts execution of the specified line of code in the library.
     */
    fun execute(code: String?)
    /**
     * Starts processing the location counter in the library.
     */
    fun executeCounter()
    val gameState: LibGameState
    val gameInterface: GameInterface
    val returnValueFlow: MutableSharedFlow<LibReturnValue>
}
